### PR TITLE
Convert definitions to resources

### PR DESCRIPTION
**Pull request by ramereth**
This includes a variety of fixes most of which gets this ready for Chef 16 properly by converting the definitions to resources.

## Description

[Describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves]

### Check List
- [ ] All tests pass. See https://github.com/sous-chefs/apache2/blob/master/TESTING.md
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by detjensrobert — DISMISSED** [review]
LGTM

File: libraries/helpers.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def nagios_vname
         'nagios4'
       elsif platform?('ubuntu')
         case node['platform_version'].to_f
-        when 16.04, 18.04
+        when 18.04
```
Comment (tas50) [inline libraries/helpers.rb]: This whole thing could be improved to make this more future proof. Remove the case statement and just make it

```ruby
if node['platform_version'].to_f < 20.04
  'nagios3'
else 
  'nagios4'
end
```

Then it just magically works on 21.x

File: test/smoke/swappable_config/default_test.rb [hunk 2]
```diff
@@ -1,4 +1,22 @@
-describe file('/etc/nagios/nagios.cfg') do
+vname =
+  if os.name == 'debian'
+    'nagios4'
+  elsif os.name == 'ubuntu'
+    case os.release.to_f
+    when 18.04
+      'nagios3'
+    when 20.04
```
Comment (tas50) [inline test/smoke/swappable_config/default_test.rb]: This should just be > 20.04 here so these tests continue to work on newer versions

**Comment by ramereth** [comment]
FYI I'm working on addressing an issue mentioned in d57c7a674b98d78f613ecb838f57a5515dd07f0d

**Review by xorima — APPROVED** [review]

**Review by detjensrobert — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [10.0.0](https://github.com/sous-chefs/nagios/releases/tag/10.0.0)

**Pull request merged**